src/services/data-management/payload-decoder-listener.service.ts

```diff
@@ -52,7 +55,7 @@ export class PayloadDecoderListenerService extends AbstractKafkaConsumer {
         dto: RawIoTDeviceRequestDto
     ) {
         const uniqueCombinations = _.uniqBy(connections.data, x => x.payloadDecoder?.id);
-        uniqueCombinations.forEach(async connection => {
+        for (const connection of uniqueCombinations) {
             try {
                 const iotDevice = connection.iotDevices.find(
                     x => x.id == dto.iotDeviceId
```

Comment:
I guess this should be "===" :-) (even though i know it was made before this PR)